modules/gdscript/editor/gdscript_highlighter.cpp

```diff
@@ -151,10 +151,13 @@ Dictionary GDScriptSyntaxHighlighter::_get_line_syntax_highlighting_impl(int p_l
 							}
 						}
 						// "#region" and "#endregion" only highlighted if they're the first region on the line.
-						if (color_regions[c].type == ColorRegion::TYPE_CODE_REGION &&
-								str.strip_edges().split_spaces()[0] != "#region" &&
-								str.strip_edges().split_spaces()[0] != "#endregion") {
-							match = false;
+						if (color_regions[c].type == ColorRegion::TYPE_CODE_REGION) {
+							Vector<String> str_stripped_split = str.strip_edges().split_spaces();
+							if (!str_stripped_split.is_empty() &&
+									str_stripped_split[0] != "#region" &&
+									str_stripped_split[0] != "#endregion") {
+								match = false;
+							}
```

Comment:
```suggestion
							String stripped_str = str.strip_edges();
							if (!stripped_str.begins_with("#region") && !stripped_str.begins_with("#endregion")) {
								match = false;
							}
```

Reply:
This works but undoes the original fix where this bug was introduced.
For example, "#region" and "#endregion" here are highlighted, but don't actually create a region you can collapse or expand.
![image](https://github.com/user-attachments/assets/10f69f34-6e89-446e-a071-506ddff97790)
We need to check for whitespace after region/endregion to fix this.

Reply:
Yes, we can use the `maxsplit` parameter since we are only checking the 0th substring.

Or alternatively we could extract the alphanumeric identifier after the `#`. Since the current version does not take into account cases like `#region:Name`. I think we should add methods like `String.get_{ascii,unicode}_identifier(from: int = 0) -> String`.

Reply:
Is there a source for what characters should be after `#region` or is that up in the air? The only examples the docs show are with whitespace.